File: DesktopSample/Advanced/PageAndRDL/CustomDataProvider/C#/CustomDataProvider/CSVDataProvider/CsvConnection.cs
using System;
using System.Collections.Specialized;
using System.Data;
using System.Data.Common;

namespace ActiveReports.Samples.CustomDataProvider.CsvDataProvider
{
	/// <summary>
	/// .NETフレームワークのCSVデータプロバイダの<see cref="IDbConnection"/>を実装する方法を提供します。
	/// </summary>
	public sealed class CsvConnection : DbConnection
	{
		private readonly string _localizedName;

		/// <summary>
		/// <see cref="CsvConnection"/>クラスの新しいインスタンスを作成します。
		/// </summary>
		public CsvConnection()
		{
			_localizedName = "Csv";
		}

		/// <summary>
		/// <see cref="CsvConnection"/>クラスの新しいインスタンスを作成します。
		/// </summary>
		///<param name="localizeName"><see cref="CsvConnection"/>のインスタンスのローカライズされた名前。</param>
		public CsvConnection(string localizeName)
		{
			_localizedName = localizeName;
		}

		/// <summary>
		/// データソースへの接続を開くために使用される文字列を取得または設定します。
		/// </summary>
		/// <remarks>Csvデータプロバイダに対する使用されません。</remarks>
		public override string ConnectionString
		{
			get { return string.Empty; }
			set { }
		}

		/// <summary>
		/// 接続の実行を終了して、エラーを生成するまでの待機時間を取得します。
		/// </summary>
		/// <remarks>Csvデータプロバイダに対する使用されません。</remarks>
		public override int ConnectionTimeout
		{
			get { throw new NotImplementedException(); }
		}

		/// <summary>
		///データソースのトランザクションを開始します。
		/// </summary>
		/// 新しいトランザクションを表すオブジェクト。
		/// <remarks>Csvデータプロバイダに対する使用されません。</remarks>
		protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel)
		{
			return null;
		}

		/// <summary>
		/// データソース接続を開きます。
		/// </summary>
		/// <remarks>Csvデータプロバイダに対する使用されません。</remarks>
		public override void Open()
		{
		}

		/// <summary>
		/// データソースへの接続を閉じます。
		/// </summary>
		public override void Close()
		{
			Dispose();
		}

		protected override DbCommand CreateDbCommand()
		{
			return new CsvCommand(string.Empty);
		}

		/// <summary>
		///<see cref="CsvCommand"/>により使用されたリソースを解放します。
		/// </summary>
		public new void Dispose()
		{
			Dispose(true);
			GC.SuppressFinalize(this);
		}

		/// <summary>
		/// <see cref="CsvConnection"/>のローカライズされた名前を取得します。
		/// </summary>
		public string LocalizedName
		{
			get { return _localizedName; }
		}

		public override ConnectionState State
		{
			get { return ConnectionState.Open; }
		}

		public override string Database
		{
			get
			{
				return string.Empty;
			}
		}

		public override string DataSource
		{
			get
			{
				throw new NotImplementedException();
			}
		}

		public override string ServerVersion
		{
			get
			{
				throw new NotImplementedException();
			}
		}

		/// <summary>
		/// この拡張機能の構成情報を指定します。
		/// </summary>
		/// <param name="configurationSettings">この設定の<see cref="NameValueCollection"/>。</param>
		public void SetConfiguration(NameValueCollection configurationSettings)
		{
		}

		public new IDbTransaction BeginTransaction(IsolationLevel il)
		{
			//do nothing
			return null;
		}

		public override void ChangeDatabase(string databaseName)
		{
			//do nothing
		}
	}
}

File: DesktopSample/Advanced/PageAndRDL/CustomDataProvider/VB.NET/CustomDataProvider/CSVDataProvider/CsvConnection.vb
Imports System.Data.Common
Imports System.Data
Imports System.Collections.Specialized
Imports System


''' <summary>
''' .NETフレームワークのCSVデータプロバイダの<see cref="IDbConnection"/>を実装する方法を提供します。
''' </summary>
Public NotInheritable Class CsvConnection
	Inherits DbConnection
	Private ReadOnly _localizedName As String

	''' <summary>
	''' <see cref="CsvConnection"/>クラスの新しいインスタンスを作成します。
	''' </summary>
	Public Sub New()
		_localizedName = "Csv"
	End Sub

	''' <summary>
	''' <see cref="CsvConnection"/>クラスの新しいインスタンスを作成します。
	''' </summary>
	'''<param name="localizeName"><see cref="CsvConnection"/>のインスタンスのローカライズされた名前。</param>
	Public Sub New(localizeName As String)
		_localizedName = localizeName
	End Sub

	''' <summary>
	''' データソースへの接続を開くために使用される文字列を取得または設定します。
	''' </summary>
	''' <remarks>Csvデータプロバイダに対する使用されません。</remarks>
	Public Overrides Property ConnectionString() As String
		Get
			Return String.Empty
		End Get
		Set
		End Set
	End Property

	''' <summary>
	''' 接続の実行を終了して、エラーを生成するまでの待機時間を取得します。
	''' </summary>
	''' <remarks>Csvデータプロバイダに対する使用されません。</remarks>
	Public Overrides ReadOnly Property ConnectionTimeout() As Integer
		Get
			Throw New NotImplementedException()
		End Get
	End Property

	''' <summary>
	'''データソースのトランザクションを開始します。
	''' </summary>
	''' 新しいトランザクションを表すオブジェクト。
	''' <remarks>Csvデータプロバイダに対する使用されません。</remarks>
	Protected Overrides Function BeginDbTransaction(isolationLevel As IsolationLevel) As DbTransaction
		Return Nothing
	End Function

	''' <summary>
	''' データソース接続を開きます。
	''' </summary>
	''' <remarks>Csvデータプロバイダに対する使用されません。</remarks>
	Public Overrides Sub Open()
	End Sub

	''' <summary>
	''' データソースへの接続を閉じます。
	''' </summary>
	Public Overrides Sub Close()
		Dispose()
	End Sub

	Protected Overrides Function CreateDbCommand() As DbCommand
		Return New CsvCommand(String.Empty)
	End Function

	''' <summary>
	'''<see cref="CsvCommand"/>により使用されたリソースを解放します。
	''' </summary>
	Public Shadows Sub Dispose()
		MyBase.Dispose(True)
		GC.SuppressFinalize(Me)
	End Sub

	''' <summary>
	''' <see cref="CsvConnection"/>のローカライズされた名前を取得します。
	''' </summary>
	Public ReadOnly Property LocalizedName() As String
		Get
			Return _localizedName
		End Get
	End Property

	Public Overrides ReadOnly Property State() As ConnectionState
		Get
			Return ConnectionState.Open
		End Get
	End Property

	Public Overrides ReadOnly Property Database() As String
		Get
			Return String.Empty
		End Get
	End Property

	Public Overrides ReadOnly Property DataSource() As String
		Get
			Throw New NotImplementedException()
		End Get
	End Property

	Public Overrides ReadOnly Property ServerVersion() As String
		Get
			Throw New NotImplementedException()
		End Get
	End Property

	''' <summary>
	''' この拡張機能の構成情報を指定します。
	''' </summary>
	''' <param name="configurationSettings">この設定の<see cref="NameValueCollection"/>。</param>
	Public Sub SetConfiguration(configurationSettings As NameValueCollection)
	End Sub

	Public Shadows Function BeginTransaction(il As IsolationLevel) As IDbTransaction
		'do nothing
		Return Nothing
	End Function

	Public Overrides Sub ChangeDatabase(databaseName As String)
		'do nothing
	End Sub
End Class
